lib/ansible/plugins/connection/docker.py

```diff
@@ -156,32 +176,24 @@ def put_file(self, in_path, out_path):
             raise AnsibleFileNotFound(
                 "file or module does not exist: %s" % in_path)
 
-        if self.can_copy_bothways:
-            # only docker >= 1.8.1 can do this natively
-            args = [ self.docker_cmd, "cp", in_path, "%s:%s" % (self._play_context.remote_addr, out_path) ]
-            args = [to_bytes(i, errors='strict') for i in args]
-            p = subprocess.Popen(args, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        out_path = pipes.quote(out_path)
+        # Older docker doesn't have native support for copying files into
+        # running containers, so we use docker exec to implement this
+        # Although docker version 1.8 and later provide support, the
+        # owner and group of the files are always set to root
+        executable = C.DEFAULT_EXECUTABLE.split()[0] if C.DEFAULT_EXECUTABLE else '/bin/sh'
```

Comment:
bcoca's patch has been merged so we can use ```play_context.executable ``` instead of C.DEFAULT_EXECUTABLE.  And the default value should be built in so no need for the final ```else /bin/sh```